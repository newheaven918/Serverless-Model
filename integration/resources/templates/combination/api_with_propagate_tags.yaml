Resources:
  MyApi:
    Type: AWS::Serverless::Api
    Properties:
      StageName: Prod
      Tags:
        Key1: Value1
        Key2: Value2
      PropagateTags: true
      GatewayResponses:
        DEFAULT_4XX:
          ResponseParameters:
            Headers:
              Access-Control-Allow-Origin: "'*'"

  MyFunction:
    Type: AWS::Serverless::Function
    Properties:
      Handler: index.handler
      Runtime: nodejs16.x
      InlineCode: |
        exports.handler = async (event, context, callback) => {
          return {
            statusCode: 200,
            body: 'Success'
          }
        }
      Events:
        RestApi:
          Type: Api
          Properties:
            RestApiId:
              Ref: MyApi
            Method: get
            Path: /iam
            Auth:
              Authorizer: AWS_IAM

        PostApi:
          Type: HttpApi
          Properties:
            ApiId:
              Ref: MyHttpApi
            Method: POST
            Path: /post

  MyHttpApi:
    Type: AWS::Serverless::HttpApi
    Properties:
      Tags:
        Tag1: value1
        Tag2: value2

Outputs:
  ApiArn:
    Description: API endpoint URL for Prod environment
    Value:
      Fn::Sub:
      - arn:${AWS::Partition}:apigateway:${AWS::Region}::/restapis/${ApiId}
      - ApiId: !Ref MyApi

  HttpApiArn:
    Description: API endpoint URL for Prod environment
    Value:
      Fn::Sub:
      - arn:${AWS::Partition}:apigateway:${AWS::Region}::/apis/${ApiId}
      - ApiId: !Ref MyHttpApi
Metadata:
  SamTransformTest: true
