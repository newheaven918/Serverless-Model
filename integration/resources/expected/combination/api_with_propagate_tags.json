[
  {
    "LogicalResourceId": "MyApiDeployment",
    "ResourceType": "AWS::ApiGateway::Deployment"
  },
  {
    "LogicalResourceId": "MyApi",
    "ResourceType": "AWS::ApiGateway::RestApi"
  },
  {
    "LogicalResourceId": "MyApiProdStage",
    "ResourceType": "AWS::ApiGateway::Stage"
  },
  {
    "LogicalResourceId": "MyFunction",
    "ResourceType": "AWS::Lambda::Function"
  },
  {
    "LogicalResourceId": "MyFunctionPostApiPermission",
    "ResourceType": "AWS::Lambda::Permission"
  },
  {
    "LogicalResourceId": "MyFunctionRestApiPermissionProd",
    "ResourceType": "AWS::Lambda::Permission"
  },
  {
    "LogicalResourceId": "MyFunctionRole",
    "ResourceType": "AWS::IAM::Role"
  },
  {
    "LogicalResourceId": "MyHttpApi",
    "ResourceType": "AWS::ApiGatewayV2::Api"
  },
  {
    "LogicalResourceId": "MyHttpApiApiGatewayDefaultStage",
    "ResourceType": "AWS::ApiGatewayV2::Stage"
  }
]
